application/btsieve/src/route_factory.rs

```diff
@@ -51,16 +47,23 @@ impl RouteFactory {
     }
 
     pub fn create<
-        Q: QueryType + ExpandResult + ShouldExpand + DeserializeOwned + Serialize + Send + 'static,
+        R,
+        Q: QueryType + DeserializeOwned + Serialize + Debug + Send + 'static,
         QR: QueryRepository<Q>,
         QRR: QueryResultRepository<Q>,
+        C: 'static + Send + Sync,
     >(
         &self,
         query_repository: Arc<QR>,
         query_result_repository: Arc<QRR>,
-        client: Option<Arc<<Q as ExpandResult>::Client>>,
+        client: Arc<C>,
         ledger_name: &'static str,
-    ) -> BoxedFilter<(impl Reply,)> {
+    ) -> BoxedFilter<(impl Reply,)>
+    where
+        for<'de> R: Deserialize<'de>,
```

Comment:
❓ `where for` -- what is this?

Reply:
Has something to do with lifetimes and allows you do specify `'de` for the `Deserialize` trait without declaring it for the local fn. I never write this myself but just let rustc tell me to do it and copy it from the error :D

Reply:
I thought you were better than that @thomaseizinger  😄